packages/backend-storage/src/policy_arbiter.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  ConstructFactoryGetInstanceProps,
+  ResourceAccessAcceptor,
+  SsmEnvironmentEntriesGenerator,
+} from '@aws-amplify/plugin-types';
+import { AccessGenerator } from './types.js';
+import { IBucket } from 'aws-cdk-lib/aws-s3';
+import { storageAccessBuilder } from './access_builder.js';
+import { BucketPolicyFactory, Permission } from './policy_factory.js';
+
+/**
+ * Middleman between creating bucket policies and attaching those policies to corresponding roles
+ */
+export class BucketPolicyArbiter {
+  /**
+   * Instantiate with context from the storage factory
+   */
+  constructor(
+    private readonly name: string,
+    private readonly accessGenerator: AccessGenerator,
+    private readonly ssmEnvironmentEntriesGenerator: SsmEnvironmentEntriesGenerator,
+    private readonly getInstanceProps: ConstructFactoryGetInstanceProps,
+    private readonly bucket: IBucket,
+    private readonly bucketPolicyFactory = new BucketPolicyFactory(bucket)
+  ) {}
+
+  /**
+   * Responsible for evaluating the access definition,
+   * creating bucket policies corresponding to the definition,
+   * then invoking the corresponding resource access acceptor to accept the policies
+   */
+  arbitratePolicies = () => {
+    const accessDefinition = this.accessGenerator(storageAccessBuilder);
+
```

Comment:
This is a lot to decompress here, can you add comments here on how accessDefinition is actually built (with the indirection happening here)